weave/trace_server_bindings/remote_http_trace_server.py

```diff
@@ -188,11 +192,34 @@ def _flush_calls(
             self._flush_calls(batch[split_idx:], _should_update_batch_size=False)
             return
 
-        # Send the batch to the server with retries
-        self._send_batch_to_server(encoded_data)
+        # If a single item is too large, we can't send it - raise an error
+        if encoded_bytes > self.remote_request_bytes_limit and len(batch) == 1:
+            raise ValueError(
+                f"Single call size ({encoded_bytes} bytes) is too large to send. "
+                f"The maximum size is {self.remote_request_bytes_limit} bytes."
+            )
+
+        try:
+            self._send_batch_to_server(encoded_data)
+        except Exception:
+            # Add items back to the queue for later processing
+            logger.warning(
+                f"Batch failed after max retries, requeueing batch with {len(batch)=} for later processing",
+            )
+
+            # only if debug mode
+            if logger.isEnabledFor(logging.DEBUG):
+                ids = []
+                for item in batch:
+                    if isinstance(item, StartBatchItem):
+                        ids.append(f"{item.req.start.id}-start")
+                    elif isinstance(item, EndBatchItem):
+                        ids.append(f"{item.req.end.id}-end")
+                logger.debug(f"Requeueing batch with {ids=}")
+            self.call_processor.enqueue(batch)
```

Comment:
when would this ever actually succeed after failing 3 times?

Reply:
if you had an ephemeral error